GDAX/AuthenticatedClient.py

```diff
@@ -56,16 +56,20 @@ def holdsPagination(self, accountId, list, after):
             self.holdsPagination(accountId, list, r.headers["cb-after"])
         return list
 
-    def buy(self, buyParams):
-        buyParams["side"] = "buy"
-        if not buyParams["product_id"]:
-            buyParams["product_id"] = self.productId
-        r = requests.post(self.url + '/orders', data=json.dumps(buyParams), auth=self.auth)
+    def buy(self, **kwargs):
+        kwargs["side"] = "buy"
+        if not "product_id" in kwargs:
+            kwargs["product_id"] = self.productId
```

Comment:
All of this totally makes sense. The buy/sell architecture needs heavy overhaul.